### PR TITLE
BL-969-ds-research-additional-graphs

### PR DESCRIPTION
**Pull request by zoebrantley**
## Description
This new feature adds three new visualizations using data from the Underdog Devs database. With each visualization is a brief description of what the graph represents.

## Jira Ticket
[BL-969](https://bloomtechlabs.atlassian.net/browse/BL-969)

## Type of Change
- [x] New feature

## Checklist
- [x] My code follows PEP8 style guide
- [x] I have removed unnecessary print statements from my code
- [x] I have made corresponding changes to the documentation if necessary
- [x] My changes generate no errors
- [x] No commented-out code
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

## Loom Video
[Loom for BL-969](https://www.loom.com/share/5714670f7b034661abc24f60eae63408)

### CODE REVIEW

**Review by randynobx — DISMISSED** [review]
These charts look awesome! The additions to `graphs.py` and `graph_router.py` look very clean.

My only critique is the number of commits. Many of them could probably be squashed together without losing anything, while reducing the footprint in the git log once it is merged.

Excellent work all around, I look forward to seeing these graphs on the deployed site! Approved.

**Review by JeffDesch — DISMISSED** [review]
Code is readable and well-structured. Graphs provide some much-needed visualization and are well-labeled. As Randy noted, just needs some interactive rebasing to clear up extra commits.

**Review by nikolayglushetskiy — APPROVED** [review]
Nice graphs!

File: app/graphs.py [hunk 1]
```diff
@@ -45,6 +45,79 @@ def df_tech_stack_by_role(database: MongoDB) -> DataFrame:
...
+
+def df_mentor_mentee(database: MongoDB) -> DataFrame:
+    mentees = DataFrame(database.read("Mentees"))
+    mentees["role"] = "Mentee"
+    mentors = DataFrame(database.read("Mentors"))
+    mentors["role"] = "Mentor"
+    return pandas.concat([mentees, mentors])
+
+
 if __name__ == '__main__':
```
[the remark this answers is not in the document]
Comment (TophertheCoder) [reply app/graphs.py]: Done. We weren't sure if it was needed so didn't want to assume and be incorrect. Thanks for letting us know!

**Review by randynobx — DISMISSED** [review]
My above comments still apply. Great work!

**Comment by BrokenShell** [comment]
This looks good overall. One thing that remains... bump the api version. This is a non-breaking change so bump the far right number in the version. `version="0.50.1"` => `version="0.50.2"` in the api.py file.

**Review by JeffDesch — APPROVED** [review]
Graphs still look good, apparently you guys went above and beyond by adding them to the API endpoint. If we end up having many more graph functions we might want to refactor out the graph properties to keep the file from becoming too long & repetitive.
For the commit issues I found this handy guide (since you 'own' this branch don't be afraid to force push a new commit history) - https://hackernoon.com/beginners-guide-to-interactive-rebasing-346a3f9c3a6d

File: app/graphs.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (zoebrantley) [reply app/graphs.py]: Bumped the API! Thanks for the patience on that change, had a busy afternoon today.

**Review by JeffDesch — APPROVED** [review]
Same as before but with a bigger number, you are now qualified to release software products :+1:

**Review by randynobx — APPROVED** [review]
It's still a good PR...I'll allow it!

**Review by nikolayglushetskiy — APPROVED** [review]
Still good. 👍

**Pull request merged**